src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,348 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.EditCommand;
+import duke.command.ExitCommand;
+import duke.command.ListCommand;
+import duke.command.FindCommand;
+
+/**
+ * The `Parser` class represents a tool to make sense of user input.
+ * It provides methods to verify and understand user input, then create respective commands.
+ */
+public class Parser {
+
+    public enum TaskType {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND, UNKNOWN
+    }
```

Comment:
I think it could be good if this was extracted out to a class of its own

Reply:
I'm not sure whether it is a common practise to put enum as a class of its own or make it part of some other class, but sounds like a valid suggestion as enums are kind of similar to classes, thanks!